test/unit/org/apache/cassandra/auth/GrantAndRevokeTest.java

```diff
@@ -75,4 +108,282 @@ public void testWarnings() throws Throwable
+        {
+            assertUnauthorizedQuery(superuser, responseMsg, format("GRANT ALL PERMISSIONS ON KEYSPACE %s TO %s", keyspace, user));
+            DataResource keyspaceResource = DataResource.keyspace(keyspace);
+            for (Permission p : keyspaceResource.applicablePermissions())
+                maybeRejectGrant(superuser, p, responseMsg, format("GRANT %s ON KEYSPACE %s TO %s", p.name(), keyspace, user));
+
+            for (TableMetadata table : Schema.instance.getKeyspaceMetadata(keyspace).tables)
+            {
+                DataResource tableResource = DataResource.table(keyspace, table.name);
+                assertUnauthorizedQuery(superuser, responseMsg, format("GRANT ALL PERMISSIONS ON %s.\"%s\" TO %s", table.keyspace, table.name, user));
+                for (Permission p : tableResource.applicablePermissions())
+                    maybeRejectGrant(superuser, p, responseMsg, format("GRANT %s ON %s.\"%s\" TO %s", p.name(), table.keyspace, table.name, user));
+            }
+        }
+    }
+
+    @Test
+    public void testGrantOnAllKeyspaces() throws Throwable
+    {
+        // Granting either specific or ALL permissions on ALL KEYSPACES is allowed, however these permissions are
+        // effective for non-system keyspaces only. If for any reason it is necessary to modify permissions on
+        // on a system keyspace, it must be done using keyspace specific grant statements.
+        Session superuser = session(SUPERUSER);
+        superuser.execute(String.format("CREATE ROLE %s WITH LOGIN = TRUE AND password='%s'", user, pass));
+        superuser.execute(String.format("ALTER KEYSPACE %s WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}", SchemaConstants.TRACE_KEYSPACE_NAME));
+        superuser.execute("CREATE KEYSPACE user_keyspace WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}");
+        superuser.execute("CREATE TABLE user_keyspace.t1 (k int PRIMARY KEY)");
+
+        Session nonsuperuser = session(USER);
+        assertUnauthorizedQuery(nonsuperuser, "User user has no MODIFY permission on <table user_keyspace.t1> or any of its parents",
+                                "INSERT INTO user_keyspace.t1 (k) VALUES (0)");
+        /* FIXME: 'User user has no MODIFY permission on <table system_views.local_node> or any of its parents'
+        assertUnauthorizedQuery(nonsuperuser, "User user has no MODIFY permission on <table system.local> or any of its parents",
+                                "INSERT INTO system.local(key) VALUES ('invalid')");
+        */
+
+        superuser.execute(format("GRANT MODIFY ON ALL KEYSPACES TO %s", user));
+        // User now has write permission on non-system keyspaces only
+        nonsuperuser.execute("INSERT INTO user_keyspace.t1 (k) VALUES (0)");
+        /* FIXME: 'User user has no MODIFY permission on <table system_views.local_node> or any of its parents'
```

Comment:
@tiagomlalves I am not seeing the problem here. If the tables names between OSS and CC changed why not just change the assertion message accordingly. What am I missing?

Reply:
The problem is that we're trying to insert into `system.local` table but error message reports error on `system_views.local_node` table. Code is translating the table when writing to it but is not translating it for the error message.

Here, I see two solutions:
 1. accept the message and leave a comment with the follow up ticket
 2. use a different table with same semantics (happy to accept suggestion on the one to use).

Reply:
I am not following 100% but if you accept the current message as it is, when that bug gets eventually fixed this test will fail and the author will fix it as well. I guess there is more value in targeting sensitive ks'.